packages/calcite-components/src/components/stepper/stepper.tsx

```diff
@@ -380,17 +473,86 @@ export class Stepper implements LocalizedComponent, T9nComponent {
     });
   }
 
-  private handleDefaultSlotChange = (event: Event): void => {
-    const slottedItems = slotChangeGetAssignedElements(event);
-    const items = slottedItems.filter((el) => el?.tagName === "CALCITE-STEPPER-ITEM");
-    const spacing = Array(items.length).fill("1fr").join(" ");
-    this.el.style.gridTemplateAreas = spacing;
-    this.el.style.gridTemplateColumns = spacing;
-    this.setStepperItemNumberingSystem();
+  private renderAction(position: Position): VNode {
+    const isPositionStart = position === "start";
+    const path = isPositionStart ? "chevron-left" : "chevron-right";
+    const { currentActivePosition, singleViewMode, layout } = this;
+    const dir = getElementDir(this.el);
+    const totalItems = this.items.length;
+    const id = `${this.guid}-${isPositionStart ? "start" : "end"}`;
+
+    return layout === "horizontal" && singleViewMode ? (
+      <calcite-action
+        alignment="center"
+        appearance="transparent"
+        class={{
+          [CSS.actionIcon]: true,
+        }}
+        compact={true}
+        data-position={position}
+        disabled={
+          (currentActivePosition === 0 && isPositionStart) ||
+          (currentActivePosition === totalItems - 1 && !isPositionStart)
+        }
+        icon={path}
+        iconFlipRtl={dir === "rtl"}
+        id={id}
+        onClick={this.handleActionClick}
+        scale={this.scale}
+        text={isPositionStart ? this.messages.previousStep : this.messages.nextStep}
+      />
+    ) : null;
+  }
+
+  private handleActionClick = (event: MouseEvent): void => {
+    const currentActivePosition = this.currentActivePosition;
+    const target = event.target as HTMLCalciteActionElement;
+    if (target.getAttribute("data-position") === "start") {
+      this.prevStep();
+    } else {
+      this.nextStep();
+    }
+
+    if (
+      this.currentActivePosition &&
+      currentActivePosition !== this.currentActivePosition &&
+      !this.items[this.currentActivePosition].disabled
+    ) {
+      this.calciteStepperItemChange.emit();
+    }
   };
 
   private getFirstEnabledStepperPosition(): number {
     const enabledStepIndex = this.items.findIndex((item) => !item.disabled);
-    return enabledStepIndex > -1 ? enabledStepIndex : 0;
+
+    if (enabledStepIndex > -1) {
+      return enabledStepIndex;
+    }
+
+    return 0;
   }
+
+  private setContainerEl = (el: HTMLDivElement): void => {
+    this.containerEl = el;
+  };
+
+  handleDefaultSlotChange = (event: Event): void => {
+    const items = slotChangeGetAssignedElements(event).filter(
+      (el) => el?.tagName === "CALCITE-STEPPER-ITEM"
+    );
+    this.items = items as HTMLCalciteStepperItemElement[];
+    this.setGridTemplateColumns(items);
+    this.setStepperItemNumberingSystem();
+  };
+
+  setGridTemplateColumns = (items: Element[]): void => {
```

Comment:
This and `getMinWidthOfStepperItem` can be class functions instead of assigned bound functions since these are not event handlers and `this` is bound to the component instance at their call site. https://www.joshmorony.com/understanding-jsx-for-stencil-js-applications/#event-binding has some nice and concise info on this.